config.proto

```diff
@@ -19,8 +19,11 @@ message AgentConfig {
     // reporting holds the reporting settings for the agent
     Reporting reporting = 2;
 
-    // data_capture describes the data being captured by instrumentation
-    DataCapture data_capture = 3;
+    // data_capture lists the rules for data being captured by instrumentation.
+    // The first rule in config that matches for a path should be used if multiple rules match.
+    // Therefore the default rule (if present) should be the last rule.
+    // If no default rule is given, the default behaviour should be to allow data capture for all elements
+    repeated DataCapture data_capture = 3;
```

Comment:
I have mixed feelings from the above description. It's not vague per se, but not elegant either.

I would keep the original `data_capture` as the default one and add a new field e.g. `endpoint_data_capture` of a new type

```
message EndpointDataCapture{
   DataCapture data_capture = 1;
  // endpoint 
//  host 
// probably also the sampling config 
}
```

Reply:
We will also need whitelisted headers in the config https://github.com/hypertrace/specification/blob/main/agent/tracestate.md#whitelisted-headers. This should probably go to `DataCapture`

Reply:
Let's take whitelisting headers separately, for the purpose of core mode processing it's even ok for it to be not configurable at the moment. I've opened https://traceableai.atlassian.net/browse/ENG-8716 to track it though.